plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -5608,6 +5608,23 @@ public void testCreateTableAsSelectFromVersionedTable()
         assertUpdate("DROP TABLE " + timestampVersionedSinkTableName);
     }
 
+    @Test
+    public void testSubqueryContainVersionedTable()
+    {
+        String tableName = "test_subquery_versioned" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT 1 id", 1);
+        long snapshotId = getCurrentSnapshotId(tableName);
+        String timestamp = timestampLiteral(getCommittedAtInEpochMilliseconds(tableName, snapshotId), 9);
+        assertUpdate("INSERT INTO " + tableName + " VALUES 2", 1);
+
+        // These query fail if io.trino.sql.tree.QueryPeriod doesn't implement shallowEquals method
```

Comment:
Instead of a comment here - can we have the reason on why it was working when we implement `shallowEquals` in the commit message ?